go.mod

```diff
@@ -31,6 +31,7 @@ require (
 	go.opentelemetry.io/otel/exporters/trace/jaeger v0.13.0
 	go.opentelemetry.io/otel/sdk v0.13.0
 	golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83
+	golang.org/x/mod v0.4.2
```

Comment:
I suspect we don't want this dep in go.{mod,sum}? https://pkg.go.dev/golang.org/x/mod not sure where that got pulled in?

Reply:
This is used for [semver](https://github.com/shysank/cluster-api-provider-azure/blob/a886e433bb2543d2144b79eb6fd6a6d1ade5403b/test/e2e/aks.go#L30) parsing for calculating [aks k8s version](https://github.com/shysank/cluster-api-provider-azure/blob/a886e433bb2543d2144b79eb6fd6a6d1ade5403b/test/e2e/aks.go#L225).